core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -108,6 +140,48 @@ object PSBT extends Factory[PSBT] {
   }
 }
 
+sealed trait PSBTRecord extends NetworkElement {
+
+  def key: ByteVector
+
+  def value: ByteVector
+
+  def bytes: ByteVector = {
+    val keySize = CompactSizeUInt.calc(key)
+    val valueSize = CompactSizeUInt.calc(value)
+
+    keySize.bytes ++ key ++ valueSize.bytes ++ value
+  }
+}
+
+object PSBTRecord {
+
+  def fromBytes(bytes: ByteVector): PSBTRecord = {
+    val keyCmpctUInt = CompactSizeUInt.parse(bytes)
+
+    if (keyCmpctUInt.toLong == 0) {
+      new PSBTRecord {
+        override def key: ByteVector = ByteVector.empty
+
+        override def value: ByteVector = ByteVector.empty
+      }
+    } else {
+      val key1 = bytes.drop(keyCmpctUInt.size).take(keyCmpctUInt.toLong)
+      val valueBytes = bytes.drop(keyCmpctUInt.size + keyCmpctUInt.toLong)
+      val valueCmpctUInt = CompactSizeUInt.parse(valueBytes)
+      val value1 = valueBytes
+        .drop(valueCmpctUInt.size)
+        .take(valueCmpctUInt.toLong)
+
+      new PSBTRecord {
+        override def key: ByteVector = key1
+
+        override def value: ByteVector = value1
+      }
```

Comment:
This breaks our ADT :(
`fromBytes` should probably return `(ByteVector, ByteVector)` so that it can be used in context

Reply:
And will need a scaladoc explaining what is returned